msprime/tables.py

```diff
@@ -1235,8 +1264,27 @@ def _population_table_pickle(table):
 
 class ProvenanceTable(BaseTable):
     """
+    A table recording the provenance (i.e., history) of this table, so that the
+    origin of the underlying data and sequence of subsequent operations can be
+    traced. Each row contains a "record" string (recommended format: json) and
+    a timestamp.
+
```

Comment:
is this note about being 'provisional' still true? Should we be more prescriptive here (say it should be JSON, and include certain fields)? Probably?